src/user_lists/views.py

```diff
@@ -1,55 +1,52 @@
+class ListViewSet(viewsets.ModelViewSet):
     queryset = List.objects.filter(is_removed=False)
     serializer_class = ListSerializer
-    permission_classes = [IsAuthenticated, IsObjectOwner]
+    permission_classes = [IsAuthenticated]
 
     def get_queryset(self):
         return self.queryset.filter(created_by=self.request.user)
 
-    def create(self, request, *args, **kwargs):
-        serializer = self.get_serializer(data=request.data)
-        if not serializer.is_valid():
-            # Format validation errors into a single error message
-            error_messages = []
-            for field, errors in serializer.errors.items():
-                if isinstance(errors, list):
-                    error_messages.extend([f"{field}: {error}" for error in errors])
-                else:
-                    error_messages.append(f"{field}: {errors}")
-            error_message = " ".join(error_messages) if error_messages else "Validation error"
-            return Response({"error": error_message}, status=status.HTTP_400_BAD_REQUEST)
-        
-        self.perform_create(serializer)
-        headers = self.get_success_headers(serializer.data)
-        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
-
     def perform_create(self, serializer):
         serializer.save(created_by=self.request.user)
 
     def perform_update(self, serializer):
-        instance = serializer.save(updated_by=self.request.user)
-        _update_list_timestamp(instance, self.request.user)
+        serializer.save(updated_by=self.request.user)
 
-    def destroy(self, request, *args, **kwargs):
-        instance = self.get_object()
+    def perform_destroy(self, instance):
         instance.delete()
-        instance.items.filter(is_removed=False).delete()
-        return Response({"success": True}, status=status.HTTP_200_OK)
 
+
+class ListItemViewSet(DestroyModelMixin, viewsets.GenericViewSet):
+    queryset = ListItem.objects.filter(is_removed=False)
+    serializer_class = ListItemSerializer
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        return self.queryset.filter(parent_list__created_by=self.request.user)
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        
+        parent_list = serializer.validated_data.get('parent_list')
+        if parent_list.created_by != request.user or parent_list.is_removed:
+            return Response({"error": "Invalid list"}, status=status.HTTP_400_BAD_REQUEST)
+        
+        try:
+            serializer.save(created_by=request.user)
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        except IntegrityError:
+            return Response({"error": "Document already exists in list"}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
The validation logic should be moved into the serializer (using `validate_*` methods). This way the validation logic is within the serializer, and there won't be a need to craft error responses manually.